client/src/components/auth/Login.js

```diff
@@ -58,8 +70,9 @@ class Login extends React.Component {
     })
       .then(this.handleFetchErrors)
       .then(res => res.json())
-      .then((res) => {
-        this.props.history.push('/dashboard');
+      .then((userData) => {
+        this.props.loginUser(userData);
+        this.setState({ redirectToDashboard: true });
```

Comment:
Only log the user in if the `POST /login` request succeeds.
This can be refactored:
- move the fetch logic to the action instead of keeping it in the component
- use `async await`